### DISCUSSION TITLE
PAGFile 的 LoadAsync:completionBlock: 方法传入的 url 不变，也会触发下载

### DISCUSSION THREAD

**Discussion by littlechaochao** [Tencent/libpag#2398] [General] [answered]
PAGFile 的 LoadAsync:completionBlock: 方法传入的 url 不变，也会触发下载。可不可以设计成如果传入的是同样的 url，后传入的 url 就不触发下载了，避免性能损耗

**Comment by domchen** [comment, marked as answer]
完全相同的url是会缓存在磁盘的，看看是不是你的本地缓存满了触发了清理，或者自己有业务代码调用过 [PAGDiskCache RemoveAll]。如果同时使用了PAGImageView就会按照LRU规则进行淘汰。[PAGDiskCache SetMaxDiskSize]可以设置最大的磁盘缓存空间。

**Reply by littlechaochao** [reply]
是这种场景，就是第一个 url 的请求还在进行中，第二个相同的 url 又发了出去。希望 PAG 内部能对这种场景进行判断，如果是相同的 url，请求，如果有一个在发送，后面的就不需要发送了

**Reply by domchen** [reply]
哦，这个我们记录一下。

**Comment by kevingpqi123** [comment]
该问题已修复，可以升级至最新版本
